crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -289,9 +294,38 @@ private void checkClass(Class<?> crdClass) {
             checkForBuilderClass(crdClass, crdClass.getName() + "Fluent");
             checkForBuilderClass(crdClass, crdClass.getName() + "FluentImpl");
         }
+        if (!Modifier.isAbstract(crdClass.getModifiers())) {
+            checkForAnyGetterSetter(crdClass);
+        } else {
+            for (Class c : subtypes(crdClass)) {
+                checkForAnyGetterSetter(c);
+            }
+        }
 
     }
 
+    private void checkForAnyGetterSetter(Class<?> crdClass) {
+        boolean anyGetter = false;
+        boolean anySetter = false;
+        for (Method method : crdClass.getMethods()) {
+            if (method.isAnnotationPresent(JsonAnySetter.class)) {
+                anySetter = true;
+            }
+            if (method.isAnnotationPresent(JsonAnyGetter.class)) {
+                anyGetter = true;
+            }
+            if (anyGetter && anySetter) {
+                break;
+            }
+        }
+        if (!anySetter) {
+            err(crdClass + " lacks @JsonAnySetter");
+        }
+        if (!anyGetter) {
+            err(crdClass + " lacks @JsonAnyGetter");
+        }
+    }
```

Comment:
Technically this check verifies if at least "two" methods in the class have JsonAnySetter/Getter but what are the methods isn't important. My understanding is that we should have these annotations on additional properties setter/getter. Is that right?

Reply:
You're right that Jackson would expect those annotations on a getter and a setter. So to be 100% certain that it will work as we expect we should check that too. OTOH, this is really just trying to check that those methods weren't forgotten entirely. And it's not like we're checking for correct usage of any of the other Jackson annotations. So I'm minded to say that this weaker check is sufficient for what we're trying to achieve here, but if you disagree I can fix it.

Reply:
Yes, this code as it stands can result in false positives. Are you wanting me to fix it?

Reply:
I wanted to be sure that I didn't miss understand the objective of the changes on CRD generator. Then if you think that the fix is quite quick I would do that otherwise I have got your reason.